### DISCUSSION TITLE
How to deploy on github?

### DISCUSSION THREAD

**Discussion by dgx-spark** [ubai-m/react-portfolio#28] [Q&A] [answered]

**Comment by ubaimutl** [comment, marked as answer]
### 1. Install the  `gh-pages` using yarn 

    $ yarn add gh-pages 

### 2. Add a  `homepage`  property to the  `package.json`  file

    {
        "homepage": "https://{username}.github.io/{repo-name}",
        "name": "react-portfolio",
        "version": "0.1.0",
        "private": true,
        

### 3. Add deployment scripts to the  `package.json`  file

      "scripts": {
            "start": "react-scripts start",
            "build": "react-scripts build",
        +   "predeploy": "yarn build && cp build/index.html build/404.html",
        +   "deploy": "gh-pages -d build"

### 4. Add a "remote" that points to the GitHub repository


    $ git remote add origin https://github.com/{username}/{repo-name}.git

### 5. Deploy the React app to GitHub Pages

    $ yarn deploy
